components/server/ee/src/billing/entitlement-service-ubp.ts

```diff
@@ -107,7 +110,30 @@ export class EntitlementServiceUBP implements EntitlementService {
     }
 
     protected async hasPaidSubscription(user: User, date: Date): Promise<boolean> {
-        // TODO(gpl) UBP personal: implement!
-        return true;
+        // Member of paid team?
+        const teams = await this.teamDB.findTeamsByUser(user.id);
+        const teamSubscriptionIds = [];
+        await Promise.all(
+            teams.map(async (team) => {
+                const customer = await this.stripeService.findCustomerByTeamId(team.id);
+                if (!customer) {
+                    return;
+                }
+                const subscriptionId = await this.stripeService.findUncancelledSubscriptionByCustomer(customer.id);
+                if (subscriptionId) {
+                    teamSubscriptionIds.push(subscriptionId);
+                }
+            }),
+        );
+        if (teamSubscriptionIds.length > 0) {
+            return true;
+        }
+        // Paid user?
+        const customer = await this.stripeService.findCustomerByUserId(user.id);
```

Comment:
:thought_balloon: Can we do this the other way around: check personal first, and then go to teams? The motivation is to try to reduce the number of checks required.

:cloud: Another idea: This impl checks all teams, where we could try to return after the first positive already. Maybe there is a simple way to get there? :thinking:

Reply:
Good ideas! 👍 Done and done. ✅